src/index/hgraph_index.cpp

```diff
@@ -236,6 +236,8 @@ HGraphIndex::hnsw_add(const DatasetPtr& data) {
     auto cur_count = this->bottom_graph_->TotalCount();
     vsag::Vector<std::shared_mutex>(total + cur_count, allocator_).swap(this->neighbors_mutex_);
```

Comment:
We don't support concurrent add & search currently, only support muti-thread search. To avoid user use add & search at same time, I want to introduce a feature list for all index. only support enabled features, detail in issue #154

Reply:
background build index ?

Reply:
for concurrency, only support search